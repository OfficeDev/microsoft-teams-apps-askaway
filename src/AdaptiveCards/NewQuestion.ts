import { IAdaptiveCard } from 'adaptivecards';

/**
 * Defines the template for the adaptive card used when creating a new question.
 */
export default () =>
    <IAdaptiveCard>{
        version: '1.0.0',
        $schema: 'http://adaptivecards.io/schemas/adaptive-card.json',
        type: 'AdaptiveCard',
        body: [
            {
                type: 'TextBlock',
                text:
                    'Something went wrong submitting your question. Try again.',
                color: 'attention',
                $when: '${question != null}',
            },
            {
                type: 'TextBlock',
                text: 'Question (250 characters max)',
            },
            {
                type: 'Input.Text',
                id: 'usertext',
                placeholder: 'Type a question',
                maxLength: 250,
                isMultiline: true,
                value: '${if(question != null, question, null)}',
            },
        ],
        actions: [
            {
                id: 'submitQuestion',
                type: 'Action.Submit',
                title: 'Submit',
                data: {
                    id: 'submitQuestion',
                    amaSessionId: '${amaId}',
                },
            },
        ],
    };
