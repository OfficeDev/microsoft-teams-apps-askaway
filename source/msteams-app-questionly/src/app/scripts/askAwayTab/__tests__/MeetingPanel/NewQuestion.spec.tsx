/**
 * @jest-environment jsdom
 */

// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import NewQuestion from '../../MeetingPanel/NewQuestion';
import { Helper } from '../../shared/Helper';
import { Flex, Button, TextArea, FlexItem } from '@fluentui/react-northstar';
import { HttpService } from '../../shared/HttpService';

configure({ adapter: new enzymeAdapterReact16() });

describe('NewQuestion Component', () => {
    let httpServiceIns;
    let t;
    let activeSessionData;
    let onAddNewQuestion;

    beforeAll(() => {
        jest.mock('../../shared/HttpService');
        httpServiceIns = new HttpService();
        activeSessionData = new Helper().createEmptyActiveSessionData();
        onAddNewQuestion = jest.fn();
        t = jest.fn();
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(
            <NewQuestion t={t} teamsTabContext={{ entityId: '', locale: '' }} httpService={httpServiceIns} activeSessionData={activeSessionData} onAddNewQuestion={onAddNewQuestion} />
        );
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(
            <NewQuestion t={t} teamsTabContext={{ entityId: '', locale: '' }} httpService={httpServiceIns} activeSessionData={activeSessionData} onAddNewQuestion={onAddNewQuestion} />
        );

        expect(component.find(Flex)).toHaveLength(1);
        expect(component.find(Button)).toHaveLength(1);
        expect(component.find(TextArea)).toHaveLength(1);
        expect(component.find(FlexItem)).toHaveLength(2);
    });
});
