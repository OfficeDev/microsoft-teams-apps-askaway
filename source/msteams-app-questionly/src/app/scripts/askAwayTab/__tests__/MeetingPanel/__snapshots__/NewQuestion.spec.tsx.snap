// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion Component should match the snapshot 1`] = `
<Flex
  className="input-text-field"
  gap="gap.small"
  hAlign="center"
  vAlign="end"
>
  <FlexItem>
    <TextArea
      accessibility={[Function]}
      as="textarea"
      className="text-question"
      fluid={true}
      inverted={true}
      maxLength={250}
      onChange={[Function]}
      value=""
    />
  </FlexItem>
  <FlexItem
    push={true}
  >
    <Button
      accessibility={[Function]}
      as="button"
      icon={
        <SendIcon
          onClick={[Function]}
          size="large"
        />
      }
      iconOnly={true}
      size="medium"
      text={true}
    />
  </FlexItem>
</Flex>
`;
